src/app/pages/map-page/map-page.component.ts

```diff
@@ -94,42 +85,27 @@ export class MapPageComponent implements OnInit, AfterViewInit {
     });
   }
 
-  getLocation() {
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition((position) => {
-        this.lat = position.coords.latitude;
-        this.lng = position.coords.longitude;
-        this.currentLocation = position.coords;
-        if (!this.isCurrentLocationMarkerInitialized()) {
-          this.currentLocationMarker = addMarkerWithIcon(this.lat, this.lng, 'assets/circle-16.png');
-          this.reportService.upload('');
-        }
-        if (!this.isSelectedLocationValid()) {
-          this.setSelectedLongitude(this.currentLocation.longitude.toString());
-          this.setSelectedLatitude(this.currentLocation.latitude.toString());
-        }
-        this.centerMap();
-      }, (error) => {
-        switch (error.code) {
-          case error.PERMISSION_DENIED:
-            console.log('User denied the request for Geolocation.');
-            this.getLocation();
-            break;
-          case error.POSITION_UNAVAILABLE:
-            console.log('Location information is unavailable.');
-            break;
-          case error.TIMEOUT:
-            console.log('The request to get user location timed out.');
-            break;
-        }
-      });
-    }
+  centerChanged($event:LatLngLiteral){
+    this.event.longitude = $event.lng;
+    this.event.latitude = $event.lat;
   }
 
-  initializeCurrentLocation() {
-    this.currentLocation = this.reportService.getCurrentLocationCoordinates();
-    this.lat = this.reportService.getCurrentLocationCoordinates().latitude;
-    this.lng = this.reportService.getCurrentLocationCoordinates().longitude;
+  goToCurrentLocation() {
+    this.mapLocation = new Location(this.event.latitude,this.event.longitude);
```

Comment:
whats the point of initialization the same thing at one row and then initializtion again in the next row of the same function?